### PR TITLE
doc: update installation guide to v1.2.1

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Update installation guide to v1.2.1 as the vote is passed in https://github.com/oras-project/oras/pull/1563

### CODE REVIEW

File: versioned_docs/version-1.2/installation.mdx [hunk 1]
```diff
@@ -128,11 +155,17 @@ You can install oras using the following command:
...
+:::
+
 ## Verify
 
 ```shell
 $ oras version
-Version:        1.2.0
+Version:        1.2.1
 Go version:     go1.22.3
 Git commit:     dcef719e208a9b226b15bc6512ad729a7dd93270
```
Comment (qweeah) [inline versioned_docs/version-1.2/installation.mdx:170]:
```suggestion
Git commit:     a0228556766b6276010d8feb937af512e8a50808
```

**Review by qweeah — CHANGES_REQUESTED** [review]

File: versioned_docs/version-1.2/installation.mdx [hunk 2]
```diff
@@ -128,11 +155,17 @@ You can install oras using the following command:
...
+
+:::
+
 ## Verify
 
 ```shell
 $ oras version
-Version:        1.2.0
+Version:        1.2.1
 Go version:     go1.22.3
```
Comment (qweeah) [inline versioned_docs/version-1.2/installation.mdx:169]:
```suggestion
Go version:     go1.23.4
```

**Review by qweeah — APPROVED** [review]
LGTM but IANAM

File: versioned_docs/version-1.2/installation.mdx [hunk 3]
```diff
@@ -5,7 +5,7 @@ sidebar_position: 15
 
 # Installation
 
-This guide demonstrates the installation steps of ORAS CLI on different platforms.
+This guide demonstrates the installation steps of ORAS CLI `v1.2.1` on different platforms.
```
[the remark this answers is not in the document]
Comment (FeynmanZhou) [reply versioned_docs/version-1.2/installation.mdx:8]: Hi @TerryHowe , I think it does make sense to explicitly state the version number at the beginning of the installation guide. Without this information, users might not be aware of which version will be installed using Homebrew, Winget below. 

Adding a version number here will not introduce too much maintenance cost.

[the remark this answers is not in the document]
Comment (TerryHowe) [reply versioned_docs/version-1.2/installation.mdx:8]: Sounds good

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**